### DISCUSSION TITLE
Fonts become white rectangles after upgrade to most recent ImGui/SDL3/Vulkan versions and after fixing ABI changes.

### DISCUSSION THREAD

**Discussion by NostraMagister** [ocornut/imgui#7595] [New Users Build/Link/Run issues ONLY!]
I used ImGui 1.89.6 with SDL3 3.0.0 and Vulkan 1.3.250 and with a little tweak for SDL3 I had a good running 
system using the ImGui window system to display 4 simultaneous webcams (captured with FFmpeg) as textures. 

I upgraded ImGui 1.89.6 to 1.90.6, SDL3 3.0.0 to 3.1.2 and Vulkan 1.3.250 to 1.3.283 and fixed the ABI changes.
Everything still works fine BUT I have a problem with ImGui fonts. All text characters became white square blocks 
BUT they can still be used (e.g. clicked to select web-cams) and the code continues to work correctly.

I ignored the following ImGui wiki comment because the default font worked fine before upgrading:
> 
(4) Font atlas texture fails to upload to GPU.

This is often of byproduct of point 3. If you have large number of glyphs or multiple fonts, the texture may become too big for your graphics API. The typical result of failing to upload a texture is if every glyph or everything appears as empty black or white rectangle. Mind the fact that some graphics drivers have texture size limitation. If you are building a PC application, mind the fact that your users may use hardware with lower limitations than yours.

Based on the following ImGui change log comment I made some changes:
- Backends: Vulkan: Removed parameter from ImGui_ImplVulkan_CreateFontsTexture(): backend now creates its own
   command-buffer to upload fonts. Removed ImGui_ImplVulkan_DestroyFontUploadObjects() which is now unnecessary.
   No need to call ImGui_ImplVulkan_CreateFontsTexture() as it is done automatically in ImGui_ImplVulkan_NewFrame().
   You can call ImGui_ImplVulkan_CreateFontsTexture() manually if you need to reload the font atlas texture.
   (#6943, #6715, #6327, #3743, #4618)
   
I had the following code below BEFORE upgrade. The UploadVulkanFonts() method was called WITHOUT first calling this->LoadFonts() (which contain the ImGui's io.Fonts->AddFontDefault() or io.Fonts->AddFontFromFileTTF(...)). That resulted in the default font to be properly loaded anyway. I assumed that it was present standard in the ImGui Font Altas.

// Upload Vulkan backend Fonts into the GPU
void MainWindow::UploadVulkanFonts()
{
    VkResult err = VK_SUCCESS;

    // Use any command queue and reset it so that we can start a new command buffer sequence
    VkCommandPool command_pool=this->gui_MainWindowData.Frames[this->gui_MainWindowData.FrameIndex].CommandPool;
    VkCommandBuffer command_buffer=this->gui_MainWindowData.Frames[this->gui_MainWindowData.FrameIndex].CommandBuffer;

    Check_VkResult(vkResetCommandPool(this->sdl.vulkan.vul_Device, command_pool, 0));

    // Start a new command buffer
    VkCommandBufferBeginInfo begin_info = {};
    begin_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
    begin_info.flags |= VK_COMMAND_BUFFER_USAGE_ONE_TIME_SUBMIT_BIT;
    Check_VkResult(vkBeginCommandBuffer(command_buffer, &begin_info));

    // Let ImGui add commands to the command buffer to create the font textures for loaded fonts.
    // They must have been loaded with this->LoadFonts() BEFORE calling this function.
    ImGui_ImplVulkan_CreateFontsTexture(command_buffer);
    
    // Finish the command buffer 
    VkSubmitInfo end_info = {};
    end_info.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
    end_info.commandBufferCount = 1;
    end_info.pCommandBuffers = &command_buffer;
    Check_VkResult(vkEndCommandBuffer(command_buffer));

    // Submit the command buffer and wait until all commands were processed. So we are sure not 
    // to start using fonts that are still being loaded to the GPU.
    Check_VkResult(vkQueueSubmit(this->sdl.vulkan.vul_Queue, 1, &end_info, VK_NULL_HANDLE));
    Check_VkResult(this->sdl.vulkan.DeviceWaitIdel());
    
    // Let ImGui clean-up all objects/structures it used internally and that needed to exist while the command buffer
    // was processed (because they were referenced in commands).
    ImGui_ImplVulkan_DestroyFontUploadObjects();
}

Because ImGui_ImplVulkan_CreateFontsTexture() does not need to be called anymore 
according the change log and ImGui_ImplVulkan_DestroyFontUploadObjects() doesn't 
exist anymore, the REST/REMAINDER of the above method makes no sense anymore because there is 
nothing in the command_buffer anymore. The pervious version of 
ImGui_ImplVulkan_CreateFontsTexture(command_buffer) took the buffer as an argument
and filled it.

So I simply changed the above function with a no-op:

void MainWindow::UploadVulkanFonts(){return;}

It doesn't work and shows white cubes as the characters. Then, I also called ImGui's io.Fonts->AddFontDefault() and it doesn't work either, the cubes remain. The function DOES return a font handle for the default font. I assumed that ImGui_ImplVulkan_NewFrame() would load that font. 

Can anyone tell me if there is an extra step to do to make the characters re-appear OR should this be converted in an issue?
Thanks.

**Comment by ocornut** [comment]
(For reference this has been moved to #7602)
